TestFoundation/TestCodable.swift

```diff
@@ -51,15 +51,17 @@ func expectRoundTripEquality<T : Codable>(of value: T, encode: (T) throws -> Dat
     XCTAssertEqual(value, decoded, "Decoded \(T.self) <\(decoded)> not equal to original <\(value)>")
 }
 
-func expectRoundTripEqualityThroughJSON<T : Codable>(for value: T) where T : Equatable {
+func expectRoundTripEqualityThroughJSON<T : Codable>(for value: T,
```

Comment:
Because of reasons mentioned in https://github.com/apple/swift-corelibs-foundation/pull/1135#discussion_r129784775 I don't think it makes sense putting 
```swift
encoder.nonConformingFloatEncodingStrategy = .convertToString(positiveInfinity: inf,
                                                              negativeInfinity: negInf,
                                                              nan: nan)
decoder.nonConformingFloatDecodingStrategy = .convertFromString(positiveInfinity: inf,
                                                                negativeInfinity: negInf,
                                                                nan: nan)
```

inside `expectRoundTripEqualityThroughJSON`, forcing all checks, even the ones that don't require it, to use it.

wdyt?